### DISCUSSION TITLE
for await loop with eslint

### DISCUSSION THREAD

**Discussion by halitaydin** [eslint/eslint#14247] [Q&A] [answered]
### How can I implement this for await loop with eslint?
Eslint warn me '_iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations._' when i use for await loop. (no-restricted-syntax)

Note: The code works fine with the for await loop.

```javascript
const range = {
  from: 1,
  to: 5,

  [Symbol.asyncIterator]() { // (1)
    return {
      current: this.from,
      last: this.to,

      async next() { // (2)
        // note: we can use "await" inside the async next:
        await new Promise((resolve) => setTimeout(resolve, 1000)); // (3)

        if (this.current <= this.last) {
          const result = { done: false, value: this.current };
          this.current += 1;
          return result;
        }
        return { done: true };
      },
    };
  },
};

(async () => {
  for await (const value of range) { // (4)
    console.log(value); // 1,2,3,4,5
  }
})();
```

**Comment by mdjermanovic** [comment, marked as answer]
Hi @halitaydin!

ESLint by default doesn't disallow for-await-of loops, your ESLint configuration is set to disallow them.

> Eslint warn me '_iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations._' when i use for await loop. (no-restricted-syntax)

By this [message](https://github.com/airbnb/javascript/blob/711aeb650db14c3b8cc329511ac508ee05bc1af6/packages/eslint-config-airbnb-base/rules/style.js#L347), it seems that you're using `eslint-config-airbnb`, so this would be a question for config's repository:

https://github.com/airbnb/javascript

**Reply by halitaydin** [reply]
Thank you, I'll ask them.
